web/web.go

```diff
@@ -61,7 +61,7 @@ func Handle404(a app.AppIface, w http.ResponseWriter, r *http.Request) {
 	err := model.NewAppError("Handle404", "api.context.404.app_error", nil, "", http.StatusNotFound)
 	ipAddress := utils.GetIPAddress(r, a.Config().ServiceSettings.TrustedProxyIPHeader)
 	mlog.Debug("not found handler triggered", mlog.String("path", r.URL.Path), mlog.Int("code", 404), mlog.String("ip", ipAddress))
-
+	w.Header().Set("Content-Type", "application/json")
```

Comment:
This should be moved into the `if IsAPICall(a, r) {` block. The other paths don't return JSON.